custom_components/wyzeapi/coordinator.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import binascii
+import logging
+from datetime import datetime, timedelta
+from typing import Dict
+
+from bleak import BleakClient
+from bleak_retry_connector import establish_connection
+
+from homeassistant.components import bluetooth
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import YDBLE_LOCK_STATE_UUID, YDBLE_UART_RX_UUID, YDBLE_UART_TX_UUID
+from .ydble_utils import decrypt_ecb, pack_l1, pack_l2_dict, pack_l2_lock_unlock, parse_l1, parse_l2_dict
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WyzeLockBoltCoordinator(DataUpdateCoordinator):
+    """Manages fetching data from BLE periodically."""
+
+    def __init__(self, hass, lock_service, lock) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Wyze Lock State Updater",
+            update_interval=timedelta(seconds=300),
+        )
+        self._lock_service = lock_service
+        self._lock = lock
+        # The `mac` in the original response should be UUID. 
+        # The actual MAC address should be retrieved from another API.
+        self._uuid = lock.mac
+        self._mac = None
+        self._bleak_client = None
+        self._current_command = None
+        
+    async def update_lock_info(self):
```

Comment:
Does this need to be decorated with `token_exception_handler` since its making an API call?